wasswa-derick/NairobiJavaGeeks
# [Chore #155965911] Add pre-commit hook

#### What does this PR do?
* Add a Pre-commit hook

#### Description of Task to be completed?
* Add a unit test, clean build,  java code style checks & lint checks  pre-commit hook

#### How should this be manually tested?
* Clone the project, ```git clone https://github.com/wasswa-derick/NairobiJavaGeeks.git```.
* In your terminal, ```cd NairobiJavaGeeks```
* Checkout to the ```ch-checks-precommit-hook-156005187``` branch.
* While in the project directory, open the git hooks directory by running ```open .git/hooks/```.
* Create another executable file named ```pre-commit``` and paste in the following code snippets and save.
> `# #!/bin/bash`
> `CMD="./gradlew check"`
> `./gradlew test`
> `./gradlew clean build`
> `$CMD`
> `RESULT=$?`

> `if [ $RESULT -ne 0 ] ; then`
          `echo "Static analysis problems found."`
    `exit 1`
> `fi`
> `echo "Static analysis found no problems."`
> `exit 0`

* Make the `pre-commit` script execute by running ```chmod +x .git/hooks/pre-commit```
* Change some code snippets and attempt to commit those changes.

#### What are the relevant pivotal tracker stories?
[#155965911](https://www.pivotaltracker.com/story/show/155965911)

#### ScreenShots

<img width="1370" alt="screen shot 2018-09-25 at 15 39 25" src="https://user-images.githubusercontent.com/39955231/46014888-70573600-c0d9-11e8-9f57-adeaffe9dcde.png">

<img width="637" alt="screen shot 2018-09-25 at 15 39 46" src="https://user-images.githubusercontent.com/39955231/46014891-70573600-c0d9-11e8-97e1-b5a35d5a6018.png">

<img width="1074" alt="screen shot 2018-09-25 at 15 39 58" src="https://user-images.githubusercontent.com/39955231/46014893-70573600-c0d9-11e8-8405-b8a077b72f13.png">
